### DISCUSSION TITLE
How does `SR` class is generated?

### DISCUSSION THREAD

**Discussion by B1Z0N** [dotnet/runtime#67797] [Q&A] [answered]
There are [resource files](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/Resources/Strings.resx) and also there is this [SR class](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/SR.cs). Somehow during build it generates this:
![Screenshot_20220409_115535](https://user-images.githubusercontent.com/32279961/162564459-92f96b10-b03c-4b78-b6f6-ec66cac7c411.png).

Any links on how this is possible? I can't see the place where we are specifying the name `SR` for generation.

**Comment by danmoseley** [comment, marked as answer]
https://github.com/dotnet/arcade/blob/e7ede87875f41a9b3df898ae08da5ebc96e24f56/src/Microsoft.DotNet.Arcade.Sdk/tools/GenerateResxSource.targets#L71

https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/eng/resources.targets#L8

I believe it's those files, or at least that should give you a start. 

Why are you interested?

**Reply by B1Z0N** [reply]
Thanks a lot!

> Why are you interested?

Ah, as you may remember there's [my PR on this](https://github.com/dotnet/runtime/pull/67696) and I just wanted to understand the full picture. So, just of curiosity.
